include/nlohmann/detail/input/json_sax.hpp

```diff
@@ -548,6 +680,67 @@ class json_sax_dom_callback_parser
     }
 
   private:
+
+    template<typename Value>
+    void set_start_end_pos_for_json_value(Value& v)
```

Comment:
Can we use some SFINAE magic to have two versions of this function - one for the case that `std::is_base_of<::nlohmann::detail::json_base_class_with_start_end_markers, BasicJsonType>::value)` and if not with the latter doing nothing? Then this if is done at compile time.

Reply:
Added SFINAE with use of macros